From e0fdae22db8d3a333d5c01e64d0c063b3d47c553 Mon Sep 17 00:00:00 2001
From: Sebastian Ott <sebott@redhat.com>
Date: Thu, 22 Aug 2024 16:30:08 +0200
Subject: [PATCH 6/9] x86: remove deprecated rhel machine types

RH-Author: Sebastian Ott <sebott@redhat.com>
RH-MergeRequest: 270: RHEL10 machine types
RH-Jira: RHEL-29002 RHEL-29003 RHEL-35587 RHEL-38411 RHEL-45141 RHEL-52318 RHEL-52320
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Eric Auger <eric.auger@redhat.com>
RH-Commit: [6/7] 928c59980a5c5f613e90e4ccfcfcfc08488da519 (seott1/cos-qemu-kvm)

Remove the following deprecated x86 rhel specific machine types:

pc                   RHEL 7.6.0 PC (i440FX + PIIX, 1996) (alias of pc-i440fx-rhel7.6.0)
pc-i440fx-rhel7.6.0  RHEL 7.6.0 PC (i440FX + PIIX, 1996) (default) (deprecated)
pc-q35-rhel8.6.0     RHEL-8.6.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel8.5.0     RHEL-8.5.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel8.4.0     RHEL-8.4.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel8.3.0     RHEL-8.3.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel8.2.0     RHEL-8.2.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel8.1.0     RHEL-8.1.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel8.0.0     RHEL-8.0.0 PC (Q35 + ICH9, 2009) (deprecated)
pc-q35-rhel7.6.0     RHEL-7.6.0 PC (Q35 + ICH9, 2009) (deprecated)

Signed-off-by: Sebastian Ott <sebott@redhat.com>
---
 hw/i386/pc_piix.c |  95 ----------------------------------
 hw/i386/pc_q35.c  | 128 ----------------------------------------------
 2 files changed, 223 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 663f67aa3a..96f5997051 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -864,98 +864,3 @@ static void pc_i440fx_rhel_machine_10_0_0_options(MachineClass *m)
     m->is_default = 1;
 }
 DEFINE_I440FX_MACHINE(10, 0, 0);
-
-/* Options for the latest rhel7 machine type */
-static void pc_machine_rhel7_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    m->family = "pc_piix_Y";
-    m->default_machine_opts = "firmware=bios-256k.bin,hpet=off";
-    pcmc->pci_root_uid = 0;
-    m->default_nic = "e1000";
-    m->default_display = "std";
-    m->no_parallel = 1;
-    m->numa_mem_supported = true;
-    m->auto_enable_numa_with_memdev = false;
-    machine_class_allow_dynamic_sysbus_dev(m, TYPE_RAMFB_DEVICE);
-    compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
-    m->smp_props.prefer_sockets = true;
-}
-
-static void pc_i440fx_rhel_machine_7_6_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    ObjectClass *oc = OBJECT_CLASS(m);
-    pc_machine_rhel7_options(m);
-    m->desc = "RHEL 7.6.0 PC (i440FX + PIIX, 1996)";
-    m->async_pf_vmexit_disable = true;
-    m->smbus_no_migration_support = true;
-
-    pcmc->pvh_enabled = false;
-    pcmc->default_cpu_version = CPU_VERSION_LEGACY;
-    pcmc->kvmclock_create_always = false;
-    /* From pc_i440fx_5_1_machine_options() */
-    pcmc->pci_root_uid = 1;
-    /* From pc_i440fx_7_0_machine_options() */
-    pcmc->enforce_amd_1tb_hole = false;
-    /* From pc_i440fx_8_0_machine_options() */
-    pcmc->default_smbios_ep_type = SMBIOS_ENTRY_POINT_TYPE_32;
-    /* From pc_i440fx_8_1_machine_options() */
-    pcmc->broken_32bit_mem_addr_check = true;
-    /* Introduced in QEMU 8.2 */
-    pcmc->default_south_bridge = TYPE_PIIX3_DEVICE;
-
-    object_class_property_add_enum(oc, "x-south-bridge", "PCSouthBridgeOption",
-                                   &PCSouthBridgeOption_lookup,
-                                   pc_get_south_bridge,
-                                   pc_set_south_bridge);
-    object_class_property_set_description(oc, "x-south-bridge",
-                                     "Use a different south bridge than PIIX3");
-
-    compat_props_add(m->compat_props, hw_compat_rhel_10_0,
-                     hw_compat_rhel_10_0_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_9_5,
-		     hw_compat_rhel_9_5_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_9_4,
-                     hw_compat_rhel_9_4_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_9_3,
-                     hw_compat_rhel_9_3_len);
-    compat_props_add(m->compat_props, pc_rhel_9_3_compat,
-                     pc_rhel_9_3_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_9_2,
-                     hw_compat_rhel_9_2_len);
-    compat_props_add(m->compat_props, pc_rhel_9_2_compat,
-                     pc_rhel_9_2_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_9_1,
-                     hw_compat_rhel_9_1_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_9_0,
-                     hw_compat_rhel_9_0_len);
-    compat_props_add(m->compat_props, pc_rhel_9_0_compat,
-                     pc_rhel_9_0_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_6,
-                     hw_compat_rhel_8_6_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_5,
-                     hw_compat_rhel_8_5_len);
-    compat_props_add(m->compat_props, pc_rhel_8_5_compat,
-                     pc_rhel_8_5_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_4,
-                     hw_compat_rhel_8_4_len);
-    compat_props_add(m->compat_props, pc_rhel_8_4_compat,
-                     pc_rhel_8_4_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_3,
-                     hw_compat_rhel_8_3_len);
-    compat_props_add(m->compat_props, pc_rhel_8_3_compat,
-                     pc_rhel_8_3_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_2,
-                     hw_compat_rhel_8_2_len);
-    compat_props_add(m->compat_props, pc_rhel_8_2_compat,
-                     pc_rhel_8_2_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_1, hw_compat_rhel_8_1_len);
-    compat_props_add(m->compat_props, pc_rhel_8_1_compat, pc_rhel_8_1_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_8_0, hw_compat_rhel_8_0_len);
-    compat_props_add(m->compat_props, pc_rhel_8_0_compat, pc_rhel_8_0_compat_len);
-    compat_props_add(m->compat_props, hw_compat_rhel_7_6, hw_compat_rhel_7_6_len);
-    compat_props_add(m->compat_props, pc_rhel_7_6_compat, pc_rhel_7_6_compat_len);
-}
-
-DEFINE_I440FX_MACHINE(7, 6, 0);
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 4ad4ad79a4..37f54062c8 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -760,131 +760,3 @@ static void pc_q35_rhel_machine_9_0_0_options(MachineClass *m)
 }
 
 DEFINE_Q35_MACHINE_BUGFIX(9, 0, 0);
-
-static void pc_q35_rhel_machine_8_6_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_9_0_0_options(m);
-    m->desc = "RHEL-8.6.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
-
-    pcmc->smbios_stream_product = "RHEL-AV";
-    pcmc->smbios_stream_version = "8.6.0";
-    compat_props_add(m->compat_props, hw_compat_rhel_8_6,
-                     hw_compat_rhel_8_6_len);
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 6, 0);
-
-static void pc_q35_rhel_machine_8_5_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_8_6_0_options(m);
-    m->desc = "RHEL-8.5.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
-    pcmc->smbios_stream_product = "RHEL-AV";
-    pcmc->smbios_stream_version = "8.5.0";
-    compat_props_add(m->compat_props, hw_compat_rhel_8_5,
-                     hw_compat_rhel_8_5_len);
-    compat_props_add(m->compat_props, pc_rhel_8_5_compat,
-                     pc_rhel_8_5_compat_len);
-    m->smp_props.prefer_sockets = true;
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 5, 0);
-
-static void pc_q35_rhel_machine_8_4_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_8_5_0_options(m);
-    m->desc = "RHEL-8.4.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
-    pcmc->smbios_stream_product = "RHEL-AV";
-    pcmc->smbios_stream_version = "8.4.0";
-    compat_props_add(m->compat_props, hw_compat_rhel_8_4,
-                     hw_compat_rhel_8_4_len);
-    compat_props_add(m->compat_props, pc_rhel_8_4_compat,
-                     pc_rhel_8_4_compat_len);
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 4, 0);
-
-static void pc_q35_rhel_machine_8_3_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_8_4_0_options(m);
-    m->desc = "RHEL-8.3.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
-    pcmc->smbios_stream_product = "RHEL-AV";
-    pcmc->smbios_stream_version = "8.3.0";
-    compat_props_add(m->compat_props, hw_compat_rhel_8_3,
-                     hw_compat_rhel_8_3_len);
-    compat_props_add(m->compat_props, pc_rhel_8_3_compat,
-                     pc_rhel_8_3_compat_len);
-    /* From pc_q35_5_1_machine_options() */
-    pcmc->kvmclock_create_always = false;
-    /* From pc_q35_5_1_machine_options() */
-    pcmc->pci_root_uid = 1;
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 3, 0);
-
-static void pc_q35_rhel_machine_8_2_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_8_3_0_options(m);
-    m->desc = "RHEL-8.2.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
-    m->numa_mem_supported = true;
-    m->auto_enable_numa_with_memdev = false;
-    pcmc->smbios_stream_product = "RHEL-AV";
-    pcmc->smbios_stream_version = "8.2.0";
-    compat_props_add(m->compat_props, hw_compat_rhel_8_2,
-                     hw_compat_rhel_8_2_len);
-    compat_props_add(m->compat_props, pc_rhel_8_2_compat,
-                     pc_rhel_8_2_compat_len);
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 2, 0);
-
-static void pc_q35_rhel_machine_8_1_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_8_2_0_options(m);
-    m->desc = "RHEL-8.1.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
-    pcmc->smbios_stream_product = NULL;
-    pcmc->smbios_stream_version = NULL;
-    compat_props_add(m->compat_props, hw_compat_rhel_8_1, hw_compat_rhel_8_1_len);
-    compat_props_add(m->compat_props, pc_rhel_8_1_compat, pc_rhel_8_1_compat_len);
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 1, 0);
-
-static void pc_q35_rhel_machine_8_0_0_options(MachineClass *m)
-{
-    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_rhel_machine_8_1_0_options(m);
-    m->desc = "RHEL-8.0.0 PC (Q35 + ICH9, 2009)";
-    m->smbus_no_migration_support = true;
-    m->alias = NULL;
-    pcmc->pvh_enabled = false;
-    pcmc->default_cpu_version = CPU_VERSION_LEGACY;
-    compat_props_add(m->compat_props, hw_compat_rhel_8_0, hw_compat_rhel_8_0_len);
-    compat_props_add(m->compat_props, pc_rhel_8_0_compat, pc_rhel_8_0_compat_len);
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(8, 0, 0);
-
-static void pc_q35_rhel_machine_7_6_0_options(MachineClass *m)
-{
-    pc_q35_rhel_machine_8_0_0_options(m);
-    m->alias = NULL;
-    m->desc = "RHEL-7.6.0 PC (Q35 + ICH9, 2009)";
-    m->async_pf_vmexit_disable = true;
-    compat_props_add(m->compat_props, hw_compat_rhel_7_6, hw_compat_rhel_7_6_len);
-    compat_props_add(m->compat_props, pc_rhel_7_6_compat, pc_rhel_7_6_compat_len);
-}
-
-DEFINE_Q35_MACHINE_BUGFIX(7, 6, 0);
-
-- 
2.39.3

